server/src/main/java/chat/engine/Connect.java

```diff
@@ -51,10 +49,19 @@ private BufferedReader getReader(Socket socket) throws IOException {
         InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
         return new BufferedReader(inputStreamReader);
     }
-
     private BufferedWriter getWriter(Socket socket) throws IOException {
         OutputStream outputStream = socket.getOutputStream();
         OutputStreamWriter outputStreamWriter = new OutputStreamWriter(outputStream, StandardCharsets.UTF_8);
         return new BufferedWriter(outputStreamWriter);
     }
+    private boolean disconnectFromServer(String message) throws IOException {
+        if (message.equals("exit")) {
+            LOGGER.info("Client disconnect");
+            socket.close();
+            reader.close();
+            writer.close();
```

Comment:
I see that your logic is duplicate with code in line 37-39, you should move this logic in some method, for example closeConnection